src/pb_stub.cc

```diff
@@ -812,16 +820,28 @@ Stub::SendIPCMessage(std::unique_ptr<IPCMessage>& ipc_message)
   }
 }
 
+void
+Stub::SendIPCLogMessage(std::unique_ptr<IPCMessage>& ipc_message)
+{
+  bool success = false;
+  while (!success) {
+    log_message_queue_->Push(ipc_message->ShmHandle(), 1000, success);
+  }
+}
+
 Stub::~Stub()
 {
   {
     py::gil_scoped_acquire acquire;
     model_instance_ = py::none();
   }
-
+  while (log_request_buffer.size() != 0) {
+    log_request_buffer.pop();
```

Comment:
Why do we need to pop requests here? Shouldn't the Triton process be responsible for popping messages?

Reply:
This queue is separate from the inter-process queue. This buffer only exists on the stub side and allows for a high volume of log requests to be stored without impacting performance. The thread I created reads from this buffer, processes the info and then places it on the Triton queue. Without it, the stub process would have to wait for the message synchronization to take place during the transfer.